h5py/tests/test_selections.py

```diff
@@ -105,6 +105,10 @@ def test_selection(self):
         st = sel.select((10,), list([1,2,3]), dset)
         self.assertIsInstance(st, sel.FancySelection)
 
+        # args is range, return a FancySelection
+        st = sel.select((10,), range(1, 4), dset)
+        self.assertIsInstance(st, sel.FancySelection)
```

Comment:
@tacaswell Is this sufficient to check that this matches the behavior of numpy?

Reply:
No, I think this is an internal detail.  The thing we want to track is that h5py does what ever numpy does.  There is a bunch of tests in `test_dataset_getitem.py` that test different things tat can be thrown at `__getitem__` and checks that the behavior matches numpy.

I think this probably should also get tests for more than one dimension slicing.

Reply:
Thanks for pointing me to the right file because I missed these tests. It'd be too simple otherwise.